### DISCUSSION TITLE
[Blog Plugin] RSS Feed

### DISCUSSION THREAD

**Discussion by simondueckert** [squidfunk/mkdocs-material#5822] [Q&A] [answered]
A blog has by definition an RSS feed to subscribe to it. Will the blog plugin generate a static rss file for the blog?

**Comment by squidfunk** [comment, marked as answer]
Please, [read our documentation](https://squidfunk.github.io/mkdocs-material/setup/setting-up-a-blog/#rss) before asking questions.

**Reply by simondueckert** [reply]
sry 🙈

**Comment by pchalasani** [comment]
@squidfunk I set up the rss feed following those instructions, and when I look at the page source I now see the xml links. However is there a way to show an RSS symbol on the blog? I did set up an "image" param in the yaml.

**Reply by Gregb87** [reply]
Exact same question here as pchalasani.
I have followed the instructions and can see the xml links in the page source. 
Also set an image in the yml file but didn't seem to do anything.
Would love to know how to show an RSS symbol on the page.
Thanks!

**Reply by kamilkrzyskow** [reply]
Only the rss feed in the head section is added:
https://github.com/squidfunk/mkdocs-material/blob/f1ffead06ae6858584969462563bb09ab918041f/src/templates/base.html#L63-L77

The `image` setting here is not processed by the theme:
- https://guts.github.io/mkdocs-rss-plugin/configuration/#image

You can add the icon manually, using the emoji syntax:
- https://squidfunk.github.io/mkdocs-material/reference/icons-emojis/
- `[:material-rss:](RSS_FEED_URL)`

**Reply by Gregb87** [reply]
Hi Kamil.

Thanks for that fast response.
1. Yes, I can see the RSS part in the <head> section of the base.html file in the material theme. So that part is already done.
2. The image setting - Thanks for letting me know about that not being processed by the theme. Seems like the docs at that link are a bit misleading.
3. Thanks for this tip also! Only question I have, where do you actually add the emoji syntax? I assume the mkdocs.yml file but not sure.

**Reply by kamilkrzyskow** [reply]
@Gregb87 
The `mkdocs-rss-plugin` plugin and the `mkdocs-material` theme are separate entities. The configuration at https://guts.github.io/mkdocs-rss-plugin/configuration/ only influences the generated `feeds`, the theme made it more convenient to automatically add the feed links to the `<head>` section, otherwise you'd have to add them yourself. You shouldn't think that both somehow work in unison.

Therefore, I don't think that `2.` is misleading, as it concerns adding an `<image>` tag to the RSS feed itself not the built site.
This setting might be useful to create a logo image for your feed if somebody uses some sort of aggregation tool, like `inoreader`.
- https://www.w3schools.com/xml/rss_tag_image.asp

---

The syntax I provided above `:material-rss:` is the Markdown emoji syntax, so you have to write it in a `.md` Markdown file (you also have to properly configure the `mkdocs.yml` file, as per the link in the previous comment).
- https://facelessuser.github.io/pymdown-extensions/extensions/emoji/

To give you some more ideas:

a) You can add the syntax to the `blog/index.md` file to make it appear at the top of the blog homepage. 
b) You can also use the `extra.social` section and add the icon at the bottom using a similar syntax:
- https://squidfunk.github.io/mkdocs-material/setup/setting-up-the-footer/#social-links
- Here you input the file path to an icon so in this case `material/rss`

c) You can also use the rendered `:material-rss:` HTML code and put it in some other place using template overrides:
- https://squidfunk.github.io/mkdocs-material/customization/#extending-the-theme

---

> [!TIP]
> Note that `:material-rss:` is the example I use, use the emoji search bar to find other variants.

**Reply by Gregb87** [reply]
@kamilkrzyskow 

Thanks again for the information.

I've got a few other things to take care of today. But will take this info on board and get back here with how we went setting it up.

**Reply by Gregb87** [reply]
@kamilkrzyskow 
Thanks for your help on this one.
Closing the loop and reporting that we have this working now using the extra.social option. 
The RSS feed icon shows up in the social icons area now on each page, with this yml syntax:

    - icon: material/rss
      link: /feed_rss_updated.xml
      name: Subscribe to our RSS Feed
